### PR TITLE
[fix](broadcast shuffle) fix wrong result of broadcast shuffle

### PR DESCRIPTION
**Pull request by jacktengg**
## Proposed changes

Issue Number: close #xxx

When data stream sender is doing broadcast shuffle, it accumulate to batch size and then send blocks to destinations, but  for local receivers, it ONLY send the current block, which will cause data loss.

This issue is introduced by https://github.com/apache/doris/pull/22218. 

If #22218  is pick to 2.0 branch, then also need to pick this PR.

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Review by Gabriel39 — APPROVED** [review]
LGTM

File: be/src/vec/sink/vdata_stream_sender.cpp [hunk 1]
```diff
@@ -497,19 +497,21 @@ Status VDataStreamSender::send(RuntimeState* state, Block* block, bool eos) {
...
+        Status status;                                                                            \
+        for (auto channel : _channels) {                                                          \
+            if (!channel->is_receiver_eof() && channel->is_local()) {                             \
+                status = channel->send_local_block(block);                                        \
+                HANDLE_CHANNEL_STATUS(state, channel, status);                                    \
+            }                                                                                     \
+        }                                                                                         \
         bool serialized = false;                                                                  \
         RETURN_IF_ERROR(                                                                          \
                 _serializer.next_serialized_block(block, PBLOCK, _channels.size(), &serialized)); \
```
Comment (Gabriel39) [inline be/src/vec/sink/vdata_stream_sender.cpp:509]: Do we still need to serialize this block?

**Review by Gabriel39 — CHANGES_REQUESTED** [review]

**Comment by jacktengg** [comment]
run buildall

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 45.66 seconds
 stream load tsv:          515 seconds loaded 74807831229 Bytes, about 138 MB/s
 stream load json:         20 seconds loaded 2358488459 Bytes, about 112 MB/s
 stream load orc:          65 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          30 seconds loaded 861443392 Bytes, about 27 MB/s
 insert into select:          29.3 seconds inserted 10000000 Rows, about 341K ops/s
 storage size: 17162182006 Bytes

**Comment by jacktengg** [comment]
run buildall

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 45.43 seconds
 stream load tsv:          512 seconds loaded 74807831229 Bytes, about 139 MB/s
 stream load json:         20 seconds loaded 2358488459 Bytes, about 112 MB/s
 stream load orc:          65 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          31 seconds loaded 861443392 Bytes, about 26 MB/s
 insert into select:          29.4 seconds inserted 10000000 Rows, about 340K ops/s
 storage size: 17162152778 Bytes

**Review by yiguolei — APPROVED** [review]
LGTM

**Review by Gabriel39 — APPROVED** [review]
LGTM

**Pull request merged**